### ISSUE TITLE
ReceiveMessageFromV6BoundToSpecificV4_NotReceived failed in CI on Ubuntu

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#17154]
https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu14.04_release_prtest/4644/consoleText
```
System.Net.Sockets.Tests.DualModeConnectionlessReceiveMessageFrom.ReceiveMessageFromV6BoundToSpecificV4_NotReceived [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.Net.Sockets.SocketException)
        Actual:   typeof(Xunit.Sdk.EqualException): Assert.Equal() Failure
        Expected: ::1
        Actual:   ::ffff:127.0.0.1
        Stack Trace:
  Starting:    System.Threading.Timer.Tests
  Using as the test runtime folder.
  Running tests... Start time: 13:43:32
  Commands:
  source Dumpling.sh
  /mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/bin/testhost/netcoreapp-Linux-Release-x64//dotnet xunit.console.netcore.exe System.Runtime.InteropServices.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing -notrait category=nonlinuxtests
  CollectDumps 0 System.Runtime.InteropServices.Tests
  /mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/bin/AnyOS.AnyCPU.Release/System.Runtime.InteropServices.Tests/netcoreapp /mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/src/System.Runtime.InteropServices/tests
  executing ulimit -c unlimited
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.
  
  Discovering: System.Runtime.InteropServices.Tests
           /mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs(2086,0): at System.Net.Sockets.Tests.DualModeConnectionlessReceiveMessageFrom.ReceiveMessageFrom_Helper(IPAddress listenOn, IPAddress connectTo, Boolean expectedToTimeout)
```

**Comment by karelz** [comment]
cc @Priya91 @ianhays @steveharter

**Comment by steveharter** [comment]
Assuming test bug as it is calling ReceiveMessageFrom_Helper(listenOn: IPAddress.Loopback, connectTo:IPAddress.IPv6Loopback, expectedToTimeout: true) and probably receiving another Loopback(v4) message due to port collisions on unix.

Test should be disabled on Unix along with ReceiveMessageFromV4BoundToSpecificV6_NotReceived

**Comment by karelz** [comment]
For future use: Fixed by #18576 (@steveharter can you please confirm?)
